lib/farscape/helpers/partially_ordered_list.rb

```diff
@@ -0,0 +1,75 @@
+# Convenience class that finds an order for a set of elements that satisfies an arbitrary sorting function that
+# can be undefined for some pairs.
+# TODO: Optimize and gemify (or find existing gem that does this)
+
+class PartiallyOrderedList
+  include Enumerable
+
+  class Element < Struct.new(:item, :preceders)
+    def inspect
+      if preceders.any?
+        "#{item} > {#{preceders.map(&:item).join(', ')}}"
+      else
+        item
+      end
+    end
+    def ==(other)
+      item == other.item
+    end
+  end
+
+  CircularOrderingError = Class.new(StandardError)
+
+  attr_accessor :elements, :ordering
+
+  def initialize(&block)
+    raise ArgumentError, "#{self.class}.new requires a block" unless block_given?
```

Comment:
`block_given?` is used when it is implicity included in a method.  Meaning, every ruby method accepts implicitly a block....a block does not have to formally declared in the method definitions list of parameters.

And yet, it is formally declared here.  Therefore: it should be if block.nil? , raise error.